src/os/bluestore/NVMEDevice.cc

```diff
@@ -1007,14 +1003,11 @@ int NVMEDevice::read_random(uint64_t off, uint64_t len, char *buf, bool buffered
   ++ioc.num_reading;
   driver->queue_task(t);
 
-  {
-    std::unique_lock<std::mutex> l(ioc.lock);
-    while (t->return_code > 0)
-      ioc.cond.wait(l);
+  while (t->return_code > 0) {
+	ioc.aio_wait();
```

Comment:
This logic is same with the main  branch code, it only change the ioc.cond.wait(l) with ioc.aio_wait.